src/backends/simple.ts

```diff
@@ -0,0 +1,442 @@
+            inputFileOrUrl = downloadUrl.filepath;
+            LOG && console.log('finished downloading');
+        }
+        LOG && console.log('loading', inputFileOrUrl);
+        this.demuxer = await beamcoder.demuxer('file:' + inputFileOrUrl);
+        LOG && console.log('streams', this.demuxer.streams.map(stream => stream.codecpar.codec_type));
+        this.streamIndex = this.demuxer.streams.findIndex(stream => stream.codecpar.codec_type === STREAM_TYPE_VIDEO);
+        LOG && console.log('streamIndex', this.streamIndex);
+        if (this.streamIndex === -1) {
+            throw new Error(`File has no ${STREAM_TYPE_VIDEO} stream!`);
+        }
+        LOG && console.log('stream type', this.demuxer.streams[this.streamIndex].codecpar.codec_type);
+        this.filterer = await createFilter({
+            stream: this.demuxer.streams[this.streamIndex],
+            outputPixelFormat: 'rgba',
+        });
+    }
+
+    _createDecoder() {
+        if (this.decoder) {
+            this.decoder.flush();
+            this.decoder = null;
+        }
+        this.decoder = createDecoder({
+            demuxer: this.demuxer as Demuxer,
+            streamIndex: this.streamIndex,
+            threadCount: this.threadCount,
+        });
+    }
+
+    /**
+     * Duration in seconds
+     */
+    get duration(): number {
+        const time_base = this.demuxer.streams[this.streamIndex].time_base;
+        const durations = this.demuxer.streams.map(
+            stream => stream.duration * time_base[0] / time_base[1]
+        );
+
+        return Math.max(...durations);
+    }
+
+    /**
+     * Width in pixels
+     */
+    get width(): number {
+        return this.demuxer.streams[this.streamIndex].codecpar.width;
+    }
+
+    /**
+     * Height in pixels
+     */
+    get height(): number {
+        return this.demuxer.streams[this.streamIndex].codecpar.height;
+    }
+
+    /**
+     * Get the frame for a given time in seconds
+     * @param targetTime
+     */
+    async getFrameAtTime(targetTime: number): Promise<beamcoder.Frame> {
+        LOG && console.log(`getFrameAtTime time(s)=${targetTime}`);
+        const targetPts = Math.floor(this._timeToPTS(targetTime));
+        return this._getFrameAtPts(targetPts);
+    }
+
+    /**
+     * Get imageData for a given time in seconds
+     * @param targetTime
+     */
+    async getImageDataAtTime(targetTime: number): Promise<ImageData> {
+        const targetPts = Math.floor(this._timeToPTS(targetTime));
+        LOG && console.log('targetTime', targetTime, '-> targetPts', targetPts);
+        const frame = await this._getFrameAtPts(targetPts);
+        if (!frame) {
+            LOG && console.log('no frame found');
+            return null;
+        }
+        const rawData = this._resizeFrameData(frame);
+        const image = createImageData(
+            rawData,
+            frame.width,
+            frame.height
+        ) as ImageData;
+        return image;
```

Comment:
Maybe we should consider augmenting image data with some frame metatadata, such as the PTS (and colorspace?). That would allow to write stronger tests.

Reply:
I would see 2 ways to do this:

* Add data directly in the image data
* Return an object containing the { imageData, metadata }. Probably cleaner.

Reply:
I thought this was just a temporary method. We would want to return an `Uint8ClampedArray`. @stepancar pointed out this is a more standard approach versus returning `imageData`.

Reply:
@animanathome I disagree, since Uint8ClampedArray is not so clear for any application using this data. Uint8ClampedArray could contain a .txt file, a image file, pixels of 1080x1920 or the opposite 1920x1080. It would all be the same size, but there is now way to know the image dimensions and layout pixels properly using Uint8ClampedArray.

The benefit of ImageData is that it sets a clear contract. We have this image data, with this width and height and passing it to ctx.putImageData() should work seamlessly.

Thoughts?

Reply:
Also, I think we came to an agreement with @stepancar because `getFrameAtTime` is still available, and there we have the typed array, so an application who wishes to use this can still do so!

Reply:
I'm fine with that @antoineMoPa. Do we know if there's a difference in performance? Maybe `createImageData` copies over the data? And I would suggest keeping the schema the same for `imageData`. I don't want to go to a place where we have 2 different `imageData` schemas.

Reply:
The image data is less performant because we have to trim pixels at the right. (2 for loops).

An application that works with raw data could decide to pass all data and trim the output in a more performant way (like in a shader).

Reply:
note that I don't think we need to think about it too much in the context of this PR.